src/plugins/intel_gpu/src/graph/impls/ocl/fully_connected.cpp

```diff
@@ -72,6 +81,27 @@ struct fully_connected_impl : typed_primitive_impl_ocl<fully_connected> {
 
             std::vector<layout> layouts{input0_layout, input1_layout};
 
+            bool has_zp = !primitive->compression_zero_point.empty();
+            bool has_scale = !primitive->compression_scale.empty();
+
+            size_t offset = primitive->bias.empty() ? 2 : 3;
+            const auto& weights_pshape = input1_layout.get_partial_shape();
+            if (has_zp) {
+                auto zp_layout = input_layouts[offset++];
+                if (input1_pshape.size() != 2) {
+                    zp_layout.set_partial_shape(reshape_to_2d(zp_layout.get_partial_shape(), weights_pshape[0], primitive->weights_rank));
+                }
+                layouts.push_back(zp_layout);
+            }
```

Comment:
in other parts of code, `scale` came before `zp`. Is it correct to get `zp` first from `input_layouts`?